sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -6,66 +6,56 @@ import nise from "nise";
 import {
   isBrowser,
   blobToString,
-  env,
   TestInfo,
   parseUrl,
   isPlaybackMode,
   isRecordMode,
   findRecordingsFolderPath,
   RecorderEnvironmentSetup,
-  ReplacementFunctions,
-  ReplacementMap,
   filterSecretsFromStrings,
-  filterSecretsFromJSONContent
+  filterSecretsFromJSONContent,
+  env
 } from "./utils";
 import { customConsoleLog } from "./customConsoleLog";
 
 let nock: any;
 
-let replaceableVariables: ReplacementMap;
-export function setReplaceableVariables(replacements: { [x: string]: string }): void {
-  replaceableVariables = new Map(Object.entries(replacements));
-  if (isPlaybackMode()) {
-    // Providing dummy values to avoid the error
-    for (const key of Object.keys(replacements)) {
-      env[key] = replacements[key];
-    }
-  }
-}
-
-let replacements: ReplacementFunctions = [];
-export function setReplacements(maps: any): void {
-  replacements = maps;
-}
-
-let queryParameters: string[] = [];
 /**
- *  Query Parameters, for example from the SAS token may contain sensitive information.
- *  Query Parameters provided by calling this method will be skipped.
- * @param {string[]} [params] Query Parameters to be skipped
+ * Loads the environment variables in both node and browser modes corresponding to the key-value pairs provided.
+ *
+ * Example-
+ *
+ * Suppose `replaceableVariables` is { ACCOUNT_NAME: "my_account_name", ACCOUNT_KEY: "fake_secret" },
+ * `setEnvironmentVariables` loads the ACCOUNT_NAME and ACCOUNT_KEY in the environment accordingly.
+ * @export
+ * @param {{ [key: string]: string }} replaceableVariables
  */
-export function skipQueryParams(params: string[]): void {
-  queryParameters = params;
+export function setEnvironmentVariables(replaceableVariables: { [key: string]: string }) {
+  Object.keys(replaceableVariables).map((key) => {
+    env[key] = replaceableVariables[key];
+  });
 }
 
-export function setEnvironmentOnLoad(environmentSetup: RecorderEnvironmentSetup) {
+export function setEnvironmentOnLoad() {
```

Comment:
Can we make sure that anything modified by the function is received as a parameter? 🙏 ✨

Reply:
Done